tests/test_compare_parquet_datasets.py

```diff
@@ -0,0 +1,355 @@
+from unittest import mock
+
+import pandas as pd
+from pyarrow import fs
+from moto import mock_s3
+from pandas.testing import assert_frame_equal
+
+from glue.jobs import compare_parquet_datasets as compare_parquet
+
+
+def test_that_get_duplicated_index_fields_returns_empty_df_if_no_dup_exist(
+    valid_staging_dataset,
+):
+    assert (
+        compare_parquet.get_duplicated_index_fields(
+            "dataset_fitbitactivitylogs", valid_staging_dataset
+        ).empty
+        == True
+    )
+
+
+def test_that_get_duplicated_index_fields_returns_dup_df_if_dup_exist(
+    staging_dataset_with_dup_indexes,
+):
+    assert_frame_equal(
+        compare_parquet.get_duplicated_index_fields(
+            "dataset_fitbitactivitylogs", staging_dataset_with_dup_indexes
+        ).reset_index(drop=True),
+        pd.DataFrame(
+            {
+                "LogId": ["44984262767"],
+                "StartDate": ["2021-12-24T14:27:39+00:00"],
+                "EndDate": ["2021-12-24T14:40:27+00:00"],
+            }
+        ).reset_index(drop=True),
+    )
+
+
+def test_that_get_duplicated_columns_returns_empty_if_no_dup_exist(
+    valid_staging_dataset,
+):
+    assert compare_parquet.get_duplicated_columns(valid_staging_dataset) == []
+
+
+def test_that_get_duplicated_columns_returns_list_if_dup_exist(
+    staging_dataset_with_dup_cols,
+):
+    assert compare_parquet.get_duplicated_columns(staging_dataset_with_dup_cols) == [
+        "EndDate"
+    ]
+
+
+@mock_s3
+def test_that_get_parquet_dataset_returns_empty_if_no_datasets_exist(
+    s3, mock_s3_filesystem, valid_staging_dataset, parquet_bucket_name
+):
+    data = valid_staging_dataset.to_parquet()
+    s3.create_bucket(Bucket=parquet_bucket_name)
+    s3.put_object(
+        Bucket=parquet_bucket_name,
+        Key="staging/parquet/dataset_fitbitactivitylogs/test.parquet",
+        Body=data,
+    )
+
+    file_key = "staging/parquet/dataset_fitbitactivitylogs/test.parquet"
+    parquet_dataset = compare_parquet.get_parquet_dataset(
+        dataset_key=f"{parquet_bucket_name}/{file_key}",
+        s3_filesystem=mock_s3_filesystem,
+    )
+    assert parquet_dataset == None
+
+
+@mock_s3
+def test_that_get_parquet_dataset_returns_dataset_if_datasets_exist(
```

Comment:
You might need to write a mocked class rather than use moto.

Reply:
Sadly it seems like specifically pyarrow's filesystem can't be mocked because It seems that using mock for testing purposes fails, as pyarrow (not in the most Pythonic way) checks for the type of the filesystem parameter passed to write_to_dataset: https://github.com/apache/arrow/blob/main/python/pyarrow/filesystem.py#L443-L459

Reply:
> 

So one way to approach this is to do something like:

patch.object(pq, "read_table"...)

What you care about is that a dataframe is returned from this function.

Yup, that is the plan